server/verifiers/twitter-verifier/verifier/src/main/java/org/kaaproject/kaa/server/verifiers/twitter/verifier/OAuthHeaderBuilder.java

```diff
@@ -0,0 +1,125 @@
+    private final String URL;
+    private final String ENCRYPTION_ALGO;
+    private final String CONSUMER_KEY;
+    private final String CONSUMER_SECRET;
+
+    public OAuthHeaderBuilder(String SIGNATURE_METHOD, String REQUEST_METHOD, String URL, String ENCRYPTION_ALGO,
+                              String CONSUMER_KEY, String CONSUMER_SECRET) {
+        this.SIGNATURE_METHOD = SIGNATURE_METHOD;
+        this.REQUEST_METHOD = REQUEST_METHOD;
+        this.URL = URL;
+        this.ENCRYPTION_ALGO = ENCRYPTION_ALGO;
+        this.CONSUMER_KEY = CONSUMER_KEY;
+        this.CONSUMER_SECRET = CONSUMER_SECRET;
+    }
+
+    public String generateHeader(String accessToken, String accessTokenSecret) throws InvalidKeyException {
+        String timestamp = String.valueOf(System.currentTimeMillis() / 1000);
+        String nonce = UUID.randomUUID().toString().replaceAll("-", "");
+
+        String signatureBase = generateSignatureBase(CONSUMER_KEY, accessToken, timestamp, nonce);
+        String signature = generateSignature(signatureBase, accessTokenSecret);
+
+        String header = getKeyValueString("OAuth " + OAuth.OAUTH_CONSUMER_KEY, CONSUMER_KEY,
+                OAuth.OAUTH_SIGNATURE_METHOD, SIGNATURE_METHOD,
+                OAuth.OAUTH_TIMESTAMP, timestamp,
+                OAuth.OAUTH_NONCE, nonce,
+                OAuth.OAUTH_VERSION, OAuth.VERSION_1_0,
+                OAuth.OAUTH_SIGNATURE, OAuth.percentEncode(signature),
+                OAuth.OAUTH_TOKEN, OAuth.percentEncode(accessToken));
+
+        return header;
+    }
+
+    private String generateSignatureBase(String consumerKey, String accessToken, String timestamp, String nonce) {
+        StringBuilder parameters = new StringBuilder();
+        appendQueryPairs(parameters, OAuth.OAUTH_CONSUMER_KEY, consumerKey,
+                OAuth.OAUTH_NONCE, nonce,
+                OAuth.OAUTH_SIGNATURE_METHOD, SIGNATURE_METHOD,
+                OAuth.OAUTH_TIMESTAMP, timestamp,
+                OAuth.OAUTH_TOKEN, OAuth.percentEncode(accessToken),
+                OAuth.OAUTH_VERSION, OAuth.VERSION_1_0);
+        String percentEncodedParams = OAuth.percentEncode(parameters.toString());
+        StringBuilder signatureBase = new StringBuilder();
+        appendAll(signatureBase, REQUEST_METHOD, "&", OAuth.percentEncode(URL), "&", percentEncodedParams);
+
+        return signatureBase.toString();
+    }
+
+    private String generateSignature(String signatureBase, String accessTokenSecret)
+            throws InvalidKeyException {
+        String signature = null;
+        try {
+            Mac mac = Mac.getInstance(ENCRYPTION_ALGO);
+            mac.init(new SecretKeySpec((CONSUMER_SECRET + "&" + accessTokenSecret).getBytes(), ENCRYPTION_ALGO));
+            mac.update(signatureBase.getBytes());
+            byte[] res = mac.doFinal();
+            signature = new String(Base64.encodeBase64(res)).trim();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
```

Comment:
Never do this. Propagate this exception and report is as internal appender error